src/Integration.Vsix/SonarAnalyzerManager.cs

```diff
@@ -20,15 +20,22 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Reflection;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.Shell.Interop;
 using SonarAnalyzer.Helpers;
+using SonarLint.VisualStudio.Integration.Rules;
+using SonarLint.VisualStudio.Integration.Suppression;
+using SonarLint.VisualStudio.Integration.Vsix.Suppression;
+using SonarQube.Client.Services;
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
-    public sealed class SonarAnalyzerManager
+    public sealed class SonarAnalyzerManager : IDisposable
```

Comment:
This class is probably doing too much at the moment but I couldn't find a good split so I decided to go ahead with this big class (I am open to suggestions of improvement).